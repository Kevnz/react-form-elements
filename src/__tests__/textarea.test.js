import React from 'react'
import { render, fireEvent, cleanup } from 'react-testing-library'

import { TextArea } from '../'

describe('TextBox component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should have a label', () => {
      const { getByText, getByLabelText, container } = render(
        <TextArea label="My Label" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText('My Label', { exact: false })
      const tArea = getByLabelText('My Label', { exact: false })
      expect(label.textContent).toBe('My Label')
      expect(tArea.value).toBe('')
    })
    it('should support children', () => {
      const { getByText, getByLabelText, container } = render(
        <TextArea label="My Label">This is my value</TextArea>
      )

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText('My Label', { exact: false })
      const tArea = getByLabelText('My Label', { exact: false })
      expect(label.textContent).toContain('My Label')
      expect(tArea.value).toBe('This is my value')
    })
  })
  describe('handling input', () => {
    it('should set a value when passed in', () => {
      const { getByLabelText, container } = render(
        <TextArea label="My Label" initialValue="My Input Value" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = getByLabelText('My Label', { exact: false })
      expect(input.value).toBe('My Input Value')
    })
    it('should update a value when changed', () => {
      const { getByLabelText, container } = render(
        <TextArea label="My Label">This is my value</TextArea>
      )
      const tArea = getByLabelText('My Label', { exact: false })
      fireEvent.change(tArea, { target: { value: 'Updated Value' } })
      expect(tArea.value).toBe('Updated Value')
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
