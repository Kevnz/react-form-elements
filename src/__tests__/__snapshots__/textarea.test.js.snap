// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox component The Rendered HTML should have a label 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-1-1"
  >
    My Label
  </label>
  <textarea
    id="id-1-1"
  />
</div>
`;

exports[`TextBox component The Rendered HTML should support children 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-2-1"
  >
    My Label
  </label>
  <textarea
    id="id-2-1"
  >
    This is my value
  </textarea>
</div>
`;

exports[`TextBox component handling input should set a value when passed in 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-3-1"
  >
    My Label
  </label>
  <textarea
    id="id-3-1"
  >
    My Input Value
  </textarea>
</div>
`;

exports[`TextBox component handling input should update a value when changed 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-5-1"
  >
    My Label
  </label>
  <textarea
    id="id-5-1"
  >
    Updated Value
  </textarea>
</div>
`;
